File: frontend/src/features/store/storeOperationalState.ts
export interface StoreOperationalLifecycle {
  status?: string | null
  lifecycle_status?: string | null
  operational_state?: string | null
  is_live?: boolean | null
}

export function isStoreLive(store: StoreOperationalLifecycle | null | undefined) {
  if (!store) return false
  if (store.is_live != null) return store.is_live === true
  if (store.operational_state != null) return store.operational_state.toUpperCase() === 'LIVE'
  return (store.status ?? '').toLowerCase() === 'active'
    && (store.lifecycle_status ?? '').toUpperCase() === 'ACTIVE'
}

File: backend/src/main/java/com/restaurant/system/user/StoreOperationalState.java
package com.restaurant.system.user;

import com.restaurant.system.user.entity.Store;

/** Canonical Store operational lifecycle contract. */
public final class StoreOperationalState {

    public static final String LIVE = "LIVE";
    public static final String NOT_LIVE = "NOT_LIVE";

    private StoreOperationalState() {
    }

    public static boolean isLive(Store store) {
        return store != null
            && "active".equalsIgnoreCase(store.status)
            && "ACTIVE".equalsIgnoreCase(store.lifecycle_status);
    }

    public static String value(Store store) {
        return isLive(store) ? LIVE : NOT_LIVE;
    }
}
